image-hd.c

```diff
@@ -640,6 +645,7 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 				image_error(image, "partitions exceed device size\n");
 				return -EINVAL;
 			}
+			partsize = rounddown(partsize, part->align);
```

Comment:
We may end up with a zero size partition here, so this should be moved before the check above. And change that to `partsize <= 0`.

Reply:
There's already a "generic" check for part->size being zero applied to all partitions a little later. Hm, but I suppose ending up with a size 0 here could confuse the intervening block that handles `if (part->image)` to bump part->size up to the image's size, thus exceeding the whole image's size. I'll do as you suggest.